modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PanelTag.java

```diff
@@ -0,0 +1,260 @@
+		}
+
+		jspWriter.write("</div>");
+
+		return EVAL_PAGE;
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		String panelTitleElement = "div";
+		String panelTitleCssClasses = "panel-header";
+		String panelTitleAttributes = "";
+		String panelTitleId = _getId();
+		String panelContentCssClasses = "panel-collapse";
+		boolean panelTitleIcon = false;
+
+		if (_collapsable) {
+			panelTitleElement = "button";
+			panelTitleCssClasses =
+				panelTitleCssClasses + " btn btn-unstyled panel-header-link";
+			panelContentCssClasses = panelContentCssClasses + " collapse";
+
+			if (Validator.isNotNull(_collapseClassNames)) {
+				panelTitleCssClasses =
+					panelTitleCssClasses + " " + _collapseClassNames;
+			}
+
+			if (_showCollapseIcon) {
+				panelTitleCssClasses =
+					panelTitleCssClasses +
+						" collapse-icon collapse-icon-middle";
+				panelTitleIcon = true;
+			}
+
+			if (_expanded) {
+				panelContentCssClasses = panelContentCssClasses + " show";
+			}
+			else {
+				panelTitleCssClasses = panelTitleCssClasses + " collapsed";
+			}
+
+			panelTitleAttributes =
+				panelTitleAttributes + " aria-controls=\"collapsePanel\"";
+			panelTitleAttributes = panelTitleAttributes + " aria-expanded=\"";
+			panelTitleAttributes = panelTitleAttributes + _expanded;
+			panelTitleAttributes = panelTitleAttributes + "\"";
+			panelTitleAttributes = panelTitleAttributes + " data-target=\"#";
+			panelTitleAttributes = panelTitleAttributes + panelTitleId;
+			panelTitleAttributes = panelTitleAttributes + "\"";
+			panelTitleAttributes =
+				panelTitleAttributes + " data-toggle=\"liferay-collapse\"";
+			panelTitleAttributes = panelTitleAttributes + " role=\"tab\"";
+			panelTitleAttributes = panelTitleAttributes + " type=\"button\"";
+		}
+
+		jspWriter.write("<");
+		jspWriter.write(panelTitleElement);
+		jspWriter.write(" class=\"");
+		jspWriter.write(panelTitleCssClasses);
+		jspWriter.write("\" ");
+		jspWriter.write(panelTitleAttributes);
+		jspWriter.write(">");
+
+		jspWriter.write("<span class=\"panel-title\">");
+		jspWriter.write(_displayTitle);
+		jspWriter.write("</span>");
+
+		if (panelTitleIcon) {
+			IconTag iconTag = new IconTag();
+
+			jspWriter.write("<span class=\"collapse-icon-closed\">");
+			iconTag.setSymbol("angle-right");
+			iconTag.doTag(pageContext);
+			jspWriter.write("</span>");
+
+			jspWriter.write("<span class=\"collapse-icon-open\">");
+			iconTag.setSymbol("angle-down");
+			iconTag.doTag(pageContext);
+			jspWriter.write("</span>");
+		}
+
+		jspWriter.write("</");
+		jspWriter.write(panelTitleElement);
+		jspWriter.write(">");
+
+		if (_collapsable) {
+			jspWriter.write("<div class=\"");
+			jspWriter.write(panelContentCssClasses);
+			jspWriter.write("\"");
+			jspWriter.write(" id=\"");
+			jspWriter.write(panelTitleId);
+			jspWriter.write("\"");
+			jspWriter.write(" role=\"tabpanel\">");
+		}
```

Comment:
I've tried to unify the "outputs" and run the SF, It's still working so here's the result:

<img width="887" alt="image" src="https://github.com/liferay-frontend/liferay-portal/assets/46778114/2cda1a8e-d5f0-41e2-bc31-97487f987b27">

But at the same time now, the SF is failing because of this:

<img width="763" alt="image" src="https://github.com/liferay-frontend/liferay-portal/assets/46778114/06f8142c-d7a9-4689-a575-fa3814d72488">

And even in the linked example there are multiple "outputs" so what's the criteria for splitting it? (I thought the SF would have split them correctly)

Reply:
I used a combination of suggested options 1 and 3, now it should be ok! (the local SF is green)